### PR TITLE
fix(patch): [sc-1690] Log queue growing.

### PR DESCRIPTION
**Pull request by ser-0xff**
## Description

Warning will appear if a queue size for particular endpoint will reach 1Mb

## How Has This Been Tested?

Unit tests.

## Minimal checklist:

- [X] I have performed a self-review of my own code 
- [ ] I have added `DocC` code-level documentation for any public interfaces exported by the package
- [ ] I have added unit and/or integration tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: Sources/DistributedSystem/DistributedSystem.swift [hunk 1]
```diff
@@ -960,21 +964,29 @@ public class DistributedSystem: DistributedActorSystem, @unchecked Sendable {
...
             }
         }
 
-        if let continuation {
-            continuation.yield(envelope)
+        if let res {
+            let sizeBits = (UInt64.bitWidth - 8)
+            let sizeMask = ((UInt64(1) << sizeBits) - 1)
+            let queueSize = res.queueSize.wrappingIncrementThenLoad(by: envelope.size, ordering: .releasing)
+            if (queueSize & sizeMask) > (Self.endpointQueueWarningSize << (queueSize >> sizeBits)) {
```
Comment (hassila) [inline Sources/DistributedSystem/DistributedSystem.swift:985]: This is a bit unreadable, but does this properly account for not logging continuously, but only when the target queue size is over the warning threshold?

Comment (ser-0xff) [reply Sources/DistributedSystem/DistributedSystem.swift:985]: The warning threshold is multiplied by 2 each time it was breached.
So, warning will appear on 1, 2, 4, 8... etc megabytes for each particular endpoint.

File: Sources/DistributedSystem/DistributedSystem.swift [hunk 2]
```diff
@@ -960,21 +964,29 @@ public class DistributedSystem: DistributedActorSystem, @unchecked Sendable {
...
                 return nil
             }
         }
 
-        if let continuation {
-            continuation.yield(envelope)
+        if let res {
+            let sizeBits = (UInt64.bitWidth - 8)
+            let sizeMask = ((UInt64(1) << sizeBits) - 1)
+            let queueSize = res.queueSize.wrappingIncrementThenLoad(by: envelope.size, ordering: .releasing)
```
Comment (dimlio) [inline Sources/DistributedSystem/DistributedSystem.swift:984]: Why do you need `.releasing` ordering here? Isn't `.relaxed` enough?

File: Sources/DistributedSystem/DistributedSystem.swift [hunk 3]
```diff
@@ -960,21 +964,29 @@ public class DistributedSystem: DistributedActorSystem, @unchecked Sendable {
...
             default:
                 logger.error("\(channel.remoteAddressDescription): invalid actor state \(actorInfo)")
                 return nil
             }
         }
 
-        if let continuation {
-            continuation.yield(envelope)
+        if let res {
+            let sizeBits = (UInt64.bitWidth - 8)
```
Comment (dimlio) [inline Sources/DistributedSystem/DistributedSystem.swift:982]: I'd add some comment describing what's going on here.

File: Sources/DistributedSystem/DistributedSystem.swift (hunk 2 above)
Comment (ser-0xff) [reply Sources/DistributedSystem/DistributedSystem.swift:984]: Probably will be enough, yes.

File: Sources/DistributedSystem/DistributedSystem.swift (hunk 3 above)
Comment (ser-0xff) [reply Sources/DistributedSystem/DistributedSystem.swift:982]: Done.

**Review by dimlio — APPROVED** [review]

**Pull request merged**